enhancements/update/dont-require-registry-during-boot-and-upgrade.md

```diff
@@ -0,0 +1,201 @@
+---
+title: dont-require-registry-during-boot-and-upgrade
+authors:
+- "@jhernand"
+reviewers:
+- "@avishayt"
+- "@danielerez"
+- "@mrunalp"
+- "@nmagnezi"
+- "@oourfali"
+approvers:
+- "@sdodson"
+- "@zaneb"
+- "@LalatenduMohanty"
+api-approvers:
+- "@sdodson"
+- "@zaneb"
+- "@deads2k"
+- "@JoelSpeed"
+creation-date: 2023-09-21
+last-updated: 2023-09-21
+tracking-link:
+- https://issues.redhat.com/browse/RFE-4482
+see-also:
+- https://issues.redhat.com/browse/OCPBUGS-13219
+- https://github.com/openshift/enhancements/pull/1481
+- https://github.com/openshift/cluster-network-operator/pull/1803
+replaces: []
+superseded-by: []
+---
+
+# Don't require registry during boot and upgrade
+
+## Summary
+
+Ensure that clusters don't require a registry server to boot or upgrade when
+all the required images have already been pulled.
+
+## Motivation
+
+Currently during reboots and upgrades clusters may need to contact the image
+registry servers, even if the images have already been pulled. This complicates
+things for clusters that are completely disconnected or that have an slow or
+unreliable connection to the image registry servers.
+
+### User Stories
+
+#### Boot without registry
+
+As the administrator of a cluster that has all the required images already
+pulled in all the nodes, I want to be able to reboot it without requiring
+access to a registry server.
+
+### Upgrade without registry
+
+As the administrator of a cluster that has all the required images already
+pulled in all the nodes, I want to be able to upgrade it without requiring
+access to a registry server.
+
+### Goals
+
+Ensure that clusters don't require a registry server to boot or upgrade when
+all the required images have already been pulled.
+
+### Non-Goals
+
+It is not the goal of this enhancement to add a mechanism to ensure that
+required images are available. That is the subject of the [pin and pre-load
+images](https://github.com/openshift/enhancements/pull/1481) enhancement.
+
+## Proposal
+
+### Workflow Description
+
+1. The administrator of a cluster boots a node or performs an upgrade.
+
+1. All the components of the cluster ensure that if the required images have
+been already pulled they will not try to contact the registry server.
```

Comment:
per my comment in the other EP, i guess i was expecting that this EP would go into more detail about using the image pulling/pinning API (defined in the other EP) to prepull the content.  i.e a flow like:

1) user initiates pre-upgrade image pull by choosing a release image they want to upgrade to
2) upgrade controller requests pulling+pinning of all images in the release image (incidentally how are we accessing the release image itself w/ no registry?)
3) upgrade controller confirms via status that the images are pulled+pinned
4) upgrade controller proceeds w/ the upgrade

Reply:
That detail is part of the now closed #1432. They will eventually be part of another EP. This EP is only about not blocking boots or upgrades when there is no registry and the images have already been pulled.

Reply:
>  This EP is only about not blocking boots or upgrades when there is no registry and the images have already been pulled.

I see, thanks. So this EP is "make sure we don't have any components that *need* to talk to a registry during bootstrap/upgrade(if the image is available locally or we can otherwise skip the registry request)......but we'll still be talking to a registry during bootstrap/upgrade"

and the next EP will be "utilize image prepulling/pinning to ensure the content is available locally during bootstrap/upgrade"?

Reply:
Well, yes and no :-) .

Yes, we want to make sure that no component needed for boot or upgrade requires a registry server if the images have already  been pulled. Note the "if the images have already been pulled part"; this EP doesn't say anything if the images haven't been pulled. Note also that this doesn't imply any specific way to pull those images: we would like to use the API described in #1481, but this EP should be independent of that.

No, we will not be talking to a registry during boot or upgrade because it will not be needed: the images will be already pulled and no component will require to contact the registry.

The next EP will be about using the assumption in this EP (which will be verified in CI) and the mechanism in #1481 to orchestrate an upgrade that doesn't require a registry server.

Reply:
i think we're on the same page, thanks